### PR TITLE
Doc reorg for Diataxis

### PR DESCRIPTION
**Pull request by randyh62**
Created Install directory
Created How To directory
Updates API Reference material

### CODE REVIEW

**Comment by randyh62** [comment]
This PR needs to be reviewed and approved. Any chance I could get someone to take a look?

File: docs/api/helpers.rst [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+  :description: hipSOLVER documentation and API reference library
+  :keywords: hipSOLVER, rocSOLVER, ROCm, API, documentation
+
 .. _api_helpers:
 
-****************
-Helper Functions
-****************
+******************
+Helper functions
```
Comment (tfalders) [inline docs/api/helpers.rst]: Shouldn't this be "hipsolver helper functions"?

File: docs/api/auxiliary.rst [hunk 2]
```diff
@@ -1,6 +1,11 @@
+.. meta::
+  :description: hipSOLVER documentation and API reference library
+  :keywords: hipSOLVER, rocSOLVER, ROCm, API, documentation
+
+.. _library_auxiliary:
 
 ***************************
-LAPACK Auxiliary Functions
+LAPACK auxiliary functions
```
Comment (tfalders) [inline docs/api/auxiliary.rst]: Shouldn't this be "hipsolver LAPACK auxiliary functions"?

**Review by tfalders — APPROVED** [review]
Looks good! I just have a couple minor comments to address.

**Review by jzuniga-amd — CHANGES_REQUESTED** [review]
Again, I open a PR to your branch with my suggested modifications.

File: docs/api/auxiliary.rst (hunk 2 above)
Comment (randyh62) [reply docs/api/auxiliary.rst]: Updated

File: docs/api/helpers.rst (hunk 1 above)
Comment (randyh62) [reply docs/api/helpers.rst]: Updated

**Review by jzuniga-amd — APPROVED** [review]
LGTM. 
Please squash and merge when ready.

**Comment by samjwu** [comment]
Closing and reopening this PR after adding the ci:docs-only label for only applying documentation CI on this

**Review by tfalders — APPROVED** [review]

**Pull request merged**